kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/history/view/OrderHistoryRecyclerAdapter.java

```diff
@@ -125,31 +126,28 @@ private void setSubtitle(Order item) {
         private void setOrderTitle(Order item) {
             String brand = item.getTitle();
             String delimiter = " - ";
-            if (item.getOfferType() == Order.OfferTypeEnum.SPEND) {
-                String actionText = getActionText(item);
-                switch (item.getStatus()) {
-                    case COMPLETED:
+            String actionText = getActionText(item);
+            switch (item.getStatus()) {
+                case COMPLETED:
+                    if (item.getOfferType() == OfferTypeEnum.SPEND) {
                         Spannable titleSpannable = new SpannableString(brand + delimiter);
                         titleSpannable.setSpan(new ForegroundColorSpan(colorBlue),
                             0, brand.length(),
                             Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);
                         setSpannableText(R.id.title, titleSpannable);
-
-                        setText(R.id.action_text, actionText);
                         setTextColor(R.id.action_text, colorBlue);
-                        break;
-                    case FAILED:
-                        setText(R.id.title, brand + delimiter);
-
-                        setText(R.id.action_text, actionText);
-                        setTextColor(R.id.action_text, colorRed);
-                        break;
-                    default:
-                        break;
-                }
-            } else {
-                setText(R.id.title, brand);
-                setText(R.id.action_text, "");
+                    } else {
+                        setText(R.id.title, brand);
+                    }
+                    setText(R.id.action_text, actionText);
+                    break;
+                case FAILED:
+                    setText(R.id.title, brand + delimiter);
+                    setText(R.id.action_text, actionText);
```

Comment:
setText(R.id.action_text, actionText); is called in COMPLETED: and FAILED you can just use one place for all cases